test/jdk/sun/security/x509/DNSName/LeadingPeriod.java

```diff
@@ -0,0 +1,127 @@
+ * @library /test/lib
+ * @modules java.base/sun.security.x509
+ */
+
+import java.io.*;
+import java.nio.file.*;
+import java.util.*;
+import java.security.Security;
+import java.security.cert.*;
+
+public class LeadingPeriod {
+
+    private static CertPath makeCertPath(String caStr, String targetCertStr)
+        throws CertificateException {
+        // generate certificate from cert strings
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
+
+        ByteArrayInputStream is;
+
+        is = new ByteArrayInputStream(targetCertStr.getBytes());
+        Certificate targetCert = cf.generateCertificate(is);
+
+        is = new ByteArrayInputStream(caStr.getBytes());
+        Certificate ca = cf.generateCertificate(is);
+
+        // generate certification path
+        List<Certificate> list = Arrays.asList(targetCert, ca);
+
+        return cf.generateCertPath(list);
+    }
+
+    private static PKIXParameters genParams(String caStr) throws Exception {
+        // generate certificate from cert string
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
+
+        ByteArrayInputStream is = new ByteArrayInputStream(caStr.getBytes());
+        Certificate selfSignedCert = cf.generateCertificate(is);
+
+        // generate a trust anchor
+        TrustAnchor anchor;
+        anchor = new TrustAnchor((X509Certificate) selfSignedCert, null);
+
+        Set<TrustAnchor> anchors = Collections.singleton(anchor);
+
+        PKIXParameters params = new PKIXParameters(anchors);
+
+        // disable certificate revocation checking
+        params.setRevocationEnabled(false);
+
+        // disable OCSP checker
+        Security.setProperty("ocsp.enable", "false");
+
+        // disable CRL checker
+        System.setProperty("com.sun.security.enableCRLDP", "false");
+
+        return params;
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        CertPathValidator validator = CertPathValidator.getInstance("PKIX");
+
+        // Load certs with a NameConstraint where DNS value does not begin with a period
+        Path targetFromCAWithoutPeriodPath = Paths.get(System.getProperty(
+            "test.src", "./") + "/certs/withoutLeadingPeriod/leaf.pem");
+        String targetFromCAWithoutPeriod = Files.readString(targetFromCAWithoutPeriodPath);
+
+        Path caWithoutLeadingPeriodPath = Paths.get(System.getProperty(
+            "test.src", "./") + "/certs/withoutLeadingPeriod/ca.pem");
+        String caWithoutLeadingPeriod = Files.readString(caWithoutLeadingPeriodPath);
+
+        PKIXParameters paramsForCAWithoutLeadingPeriod = genParams(caWithoutLeadingPeriod);
+        CertPath pathWithoutLeadingPeriod = makeCertPath(caWithoutLeadingPeriod,
+            targetFromCAWithoutPeriod);
+        try {
+            validator.validate(pathWithoutLeadingPeriod, paramsForCAWithoutLeadingPeriod);
+        } catch (CertPathValidatorException uoe) {
```

Comment:
You don't need try/catch here since main throws Exception this will just get propagated upwards. Same comment on line 122.